### PR TITLE
See if ioutilCopy is better

### LINKED ISSUE #8127: replace ioutils.ReadAll with io.Copy in TSDB/fileutile.go/copyfile
**Use case. Why is this important?**
in TSDB/fileutile.go/copyfile, which use ioutils.Readall read data from disk to user space and then use ioutils.WriteAll write data from user space to disk.
I think it is not efficient.

And in linux copyFileRange in go, which `performs an in-kernel copy between
       two file descriptors without the additional cost of transferring data
       from the kernel to user space and then back into the kernel.`.

And in go, io.Copy implement copyFileRange.
*“Nice to have” is not a good use case. :)*

### PR DESCRIPTION
**Pull request by nrchakradhar**
Can fix #8127

### CODE REVIEW

**Comment by nrchakradhar** [comment]
@ArthurSens Here are the changes. Let me know how to trigger benchmark runs. Thanks

**Comment by ArthurSens** [comment]
It has to be a member of the Prometheus org. I'll try to ping someone at the IIRC :slightly_smiling_face:

File: tsdb/fileutil/fileutil.go [hunk 1]
```diff
@@ -59,13 +59,20 @@ func CopyDirs(src, dest string) error {
 	return nil
 }
 
+// Making changes for running benchmark in Prometheus.
```
Comment (ArthurSens) [inline tsdb/fileutil/fileutil.go:62]: This comment is unnecessary, we would want that in case the benchmarks look good and the PR gets merged.

**Comment by roidelapluie** [comment]
We do not have benchmark for this as it seems only used in tests.

**Comment by ArthurSens** [comment]
Sorry @nrchakradhar, looks like I've been leading you on the wrong way.

What are we supposed to do here, then? Is the benchmark provided at the linked issue enough?

**Comment by brian-brazil** [comment]
If this is only used in unittests, then there's no real need to make it performant - code clarity is generally more important there.

**Comment by nrchakradhar** [comment]
@brian-brazil @roidelapluie I will check once again and if it's use is only in tests, I will close the PR.
It might help to segregate test util code. That can be a different issue and PR.

**Comment by brian-brazil** [comment]
It is only used in tests, and given that Go test code can't access tests in other modules I don't think we'd gain much here by splitting out this function to a new module.

**Comment by nrchakradhar** [comment]
Agree. Its used only by copyDirs, which is used only in compact_test.go, db_test.go and repair_test.go.
Closing the PR. Will update the same in #8127 as well.
Thanks for all the help and patience.

**Comment by nrchakradhar** [comment]
> Sorry @nrchakradhar, looks like I've been leading you on the wrong way.
> 

@ArthurSens Not at all. I learnt a thing to check the utility of the function before optimizing. I also learnt what are the different ways copy can be done.

**Pull request closed without merging**